src/connectors/hits-per-page/connectHitsPerPage.js

```diff
@@ -161,7 +161,9 @@ The first one will be picked, you should probably set only one default value`
         }
 
         this.setHitsPerPage = value =>
-          helper.setQueryParameter('hitsPerPage', value).search();
+          value === 'undefined' || value === ''
```

Comment:
I don't think the `value` could be equal to `'undefined'`. Maybe use `typeof value === 'undefined'`. But IMO we can use the falsy value of `value` for this condition. Every time a falsy value is provided we reset the `hitsPerPage`.

Reply:
As of now, yes it is equal to `'undefined'` because the value for the empty entry is `undefined` and it is used directly in the DOM and stringified in the process.

I can see to change the value for the empty entry.

Reply:
Okay my bad though 👍

Reply:
Updated the value and I changed the value for the "not in the list" value.